### DISCUSSION TITLE
[V5] Working with types in service objects

### DISCUSSION THREAD

**Discussion by patni1992** [adonisjs/core#2035] [Help] [answered]
So what I do in my controller is to use a validator, then I pass the data to a service. Validate generates types at runtime but what if I want to use that type as a parameter in my create method?

```js
const validatedData = await request.validate(someValidator)

await this.someService.create(validatedData)
```

```js
class SomeService {
// what type should someData be? 
 public async create(someData) {
    
  }
}
```

Does that mean I need to create a separate type? or what is best practice here?

**Comment by popovicieduard** [comment, marked as answer]
Hello @patni1992 ,

What I usually do is create a contract folder inside app and there create different files for my services, repositories, so for example if I have an account service I would create an `app/Contracts/Account.ts` and export the following code for the get method:
```
export interface GetAccountServiceContract {
  entityType?: EntityTypeEnum
  isActive?: boolean
  isVerified?: boolean
  page?: number
  perPage?: number
  orderBy?: OrderByEnum
}
```

and have the following code:
```
import {GetAccountServiceContract} from 'App/Contracts/Account'
class AccountService {
// what type should someData be? 
 public async get(params: GetAccountServiceContract) {
    
  }
}
```

Just make sure to add the below line inside .adonissrc.json
```
  "aliases": {
    "App": "app",
    "Contracts": "contracts",
    .....
```

**Reply by patni1992** [reply]
Got it :)! appreciate the answer thank you
